src/status_im2/contexts/syncing/standard_authentication/view.cljs

```diff
@@ -0,0 +1,47 @@
+(ns status-im2.contexts.syncing.standard-authentication.view
+  (:require
+   [quo2.core :as quo]
+   [utils.re-frame :as rf]
+   [react-native.touch-id :as biometric]
+   [utils.i18n :as i18n]
+   [status-im2.contexts.syncing.standard-authentication.enter-password :as enter-password]
+   [status-im2.contexts.syncing.standard-authentication.style :as style]
+   [react-native.core :as rn]))
+
+(defn authorize [set-code use-biometric-auth auth-success auth-fail fallback-button-label]
+  (biometric/get-supported-type
+   (fn [biometric-type]
+     (if (and use-biometric-auth biometric-type)
+       (biometric/authenticate
+        {:reason        (i18n/label :t/biometric-auth-confirm-message)
+         :on-success    (fn [response]
+                          (auth-success)
+                          (js/console.log "response" response))
+         :on-fail       (fn [error]
+                          (js/console.log "Authentication Failed. Error:" error)
+                          (auth-fail)
+                          (rf/dispatch [:show-bottom-sheet
+                                        {:content (fn [] [enter-password/view set-code])}]))
+         :options       {}})
+           ;; If biometric authentication is not supported, fallback to password-based authentication
```

Comment:
imo: I prefer to avoid comments like this and just use named functions which describe the logic 
```
(let [biometric-supported? (and use-biometric-auth biometric-type)]
 (if (not biometric-supported?)
   (biomtric-auth {...})
   (password-auth {...})
 )))
 
```